wlauto/workloads/googleplaybooks/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java

```diff
@@ -0,0 +1,362 @@
+
+        waitForPage();
+
+        result.end();
+        timingResults.put(testTag, result);
+    }
+
+    private void addNote(final String text) throws Exception {
+        Timer result = new Timer();
+        result.start();
+
+        UiObject clickable = new UiObject(new UiSelector().longClickable(true));
+        uiDevicePerformLongClick(clickable, 100);
+        UiObject addNoteButton = getUiObjectByResourceId("com.google.android.apps.books:id/add_note_button",
+                                                        "android.widget.ImageButton");
+        addNoteButton.click();
+
+        UiObject noteEditText = getUiObjectByResourceId("com.google.android.apps.books:id/note_edit_text",
+                                                        "android.widget.EditText");
+        noteEditText.setText(text);
+
+        UiObject noteMenuButton = getUiObjectByResourceId("com.google.android.apps.books:id/note_menu_button",
+                                                          "android.widget.ImageButton");
+        noteMenuButton.click();
+
+        UiObject saveButton = getUiObjectByText("Save", "android.widget.TextView");
+        saveButton.click();
+
+        waitForPage();
+
+        result.end();
+        timingResults.put("add_note", result);
+    }
+
+    private void removeNote() throws Exception {
+        Timer result = new Timer();
+        result.start();
+        UiObject clickable = new UiObject(new UiSelector().longClickable(true));
+        uiDevicePerformLongClick(clickable, 100);
+
+        UiObject removeButton = getUiObjectByResourceId("com.google.android.apps.books:id/remove_highlight_button",
+                                                        "android.widget.ImageButton");
+        removeButton.click();
+
+        UiObject confirmRemove = getUiObjectByText("Remove", "android.widget.Button");
+        confirmRemove.click();
+
+        waitForPage();
+
+        result.end();
+        timingResults.put("remove_note", result);
+    }
+
+    private void searchForWord(final String text) throws Exception {
+        getDropdownMenu();
+        Timer result = new Timer();
+        result.start();
+        UiObject search = new UiObject(
+                new UiSelector().resourceId("com.google.android.apps.books:id/menu_search"));
+        search.click();
+
+
+
+        UiObject searchText = new UiObject(
+                new UiSelector().resourceId("com.google.android.apps.books:id/search_src_text"));
+        searchText.setText(text);
+        pressEnter();
+
+        UiObject resultList = new UiObject(
+                new UiSelector().resourceId("com.google.android.apps.books:id/search_results_list"));
+
+        if (!resultList.waitForExists(viewTimeout)) {
+            throw new UiObjectNotFoundException("Could not find \"search results list view\".");
+        }
+
+        result.end();
+        timingResults.put("search_for_word", result);
+
+        sleep(3); // Pause here to let the device settle.
```

Comment:
What are we waiting for exactly?  What happens in this settle time?  Sleeps of magic numbers.